src/app/task-create/task-create.component.ts

```diff
@@ -25,15 +29,20 @@ export class TaskCreateComponent implements OnInit {
   }
 
   ngOnInit(): void {}
-
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
   onCreateTask() {
     if (this.taskForm.valid) {
       const newTask: Task = {
         ...this.taskForm.value,
         createdOn: new Date()
       };
-      this.taskService.addTask(newTask);
-      this.taskForm.reset({ status: 'pending' }); // Reset form and set default status
+      this.taskService.createTask(newTask).pipe(takeUntil(this.destroy$)).subscribe(()=> {
+        this.router.navigate(['/task-list']);
+      });
+      this.taskForm.reset({ status: 'pending' }); 
```

Comment:
would be better to reset the form once the task is created (inside subscribe before navigation)